python/jupyterlab_widgets/src/manager.ts

```diff
@@ -455,7 +455,7 @@ export class WidgetManager extends LabWidgetManager {
    */
   private _saveState(): void {
     const state = this.get_state_sync({ drop_defaults: true });
-    this._context.model.metadata.set('widgets', {
+    this._context.contentsModel?.content.metadata.set('widgets', {
```

Comment:
You should never touch the `contentsModel`. In this case you should use the `model.setMetadata`

```suggestion
    this._context.model.setMetadata('widgets', {
```